src/cleanup.rs

```diff
@@ -0,0 +1,81 @@
+use crate::Config;
+use log::warn;
+use std::sync::Arc;
+
+/// Cleanup any Docker images at the end of a block,
+/// even in case of panic.
+///
+/// This reduces the chances that Docker will assume
+/// that the images must be restarted on your next computer
+/// startup and decide to bring them up as `root`.
+///
+/// As a side-effect, all tests MUST be prefixed with
+/// `#[tokio::test(flavor = "multi_thread")]`
+pub struct Cleanup {
+    /// If `true`, cleanup is still needed.
+    is_armed: bool,
+
+    /// The container name used during `build`.
+    setup_container_name: Arc<str>,
```

Comment:
wait what? how can you/why would you reference count to a slice?

Reply:
It's technically not a slice, it's the buffer holding it. `&str` would make it a slice or `Arc<str>` makes it a slice-like refcounted data structure.